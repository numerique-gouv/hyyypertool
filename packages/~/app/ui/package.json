{
  "name": "@~/app.ui",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    "./button": {
      "default": "./src/button/index.ts"
    },
    "./button/scripts": {
      "default": "./src/button/scripts/index.ts"
    },
    "./callout": {
      "default": "./src/callout/index.ts"
    },
    "./form": {
      "default": "./src/form/index.ts"
    },
    "./hx_table": {
      "default": "./src/hx_table/index.ts"
    },
    "./loader": {
      "default": "./src/loader/index.ts"
    },
    "./menu": {
      "default": "./src/menu/index.ts"
    },
    "./notice": {
      "default": "./src/notice/index.ts"
    },
    "./plurial": {
      "default": "./src/plurial/index.ts"
    },
    "./quote": {
      "default": "./src/quote/index.ts"
    },
    "./table": {
      "default": "./src/table/index.ts"
    },
    "./visually_hidden": {
      "default": "./src/visually_hidden/index.ts"
    },
    "./*": {
      "default": "./src/*.tsx"
    }
  },
  "scripts": {
    "test": "bun test"
  },
  "dependencies": {
    "@~/app.core": "workspace:*",
    "@~/app.urls": "workspace:*",
    "hono": "4.5.11",
    "hono-slotify": "workspace:*",
    "tailwind-variants": "0.2.1"
  },
  "devDependencies": {
    "@~/config.happydom": "workspace:*",
    "@~/config.tailwindcss": "workspace:*",
    "@~/config.typescript": "workspace:*"
  }
}
