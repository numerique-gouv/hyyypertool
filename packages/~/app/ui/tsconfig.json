{
  "compilerOptions": {
    "jsx": "react-jsx",
    "jsxImportSource": "hono/jsx",
    "lib": ["DOM"],
    "outDir": "./node_modules/.cache/tsc",
    "types": ["bun-types", "hono"]
  },
  "extends": "@~/config.typescript/tsx/tsconfig.json",
  "references": [
    {
      "path": "../core"
    },
    {
      "path": "../urls"
    },
    {
      "path": "../../../hono-slotify"
    },
    {
      "path": "../../config/happydom"
    }
  ]
}
