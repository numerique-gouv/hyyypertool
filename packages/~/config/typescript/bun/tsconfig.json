{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Bun",
  "extends": "../base/tsconfig.json",
  "compilerOptions": {
    "types": ["bun-types"]
  }
}
