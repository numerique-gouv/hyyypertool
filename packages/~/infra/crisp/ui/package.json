{
  "name": "@~/crisp.ui",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    ".": {
      "default": "./src/index.ts"
    },
    "./message": {
      "default": "./src/Message/index.tsx"
    },
    "./*": {
      "default": "./src/*.tsx"
    }
  },
  "dependencies": {
    "@~/crisp.lib": "workspace:*",
    "rehype-sanitize": "6.0.0",
    "rehype-stringify": "10.0.1",
    "remark-parse": "11.0.0",
    "remark-rehype": "11.1.1",
    "ts-pattern": "5.4.0",
    "unified": "11.0.5"
  },
  "devDependencies": {
    "@~/config.tailwindcss": "workspace:*",
    "@~/config.typescript": "workspace:*"
  }
}
