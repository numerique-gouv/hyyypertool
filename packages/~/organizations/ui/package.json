{
  "name": "@~/organizations.ui",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "exports": {
    "./*": {
      "default": "./src/*.tsx"
    }
  },
  "dependencies": {
    "@~/app.ui": "workspace:*",
    "@~/app.urls": "workspace:*",
    "@~/organizations.lib": "workspace:*",
    "hono": "4.7.11",
    "tailwind-variants": "0.3.1",
    "ts-pattern": "5.7.0",
    "zod": "3.24.1"
  },
  "devDependencies": {
    "@~/config.typescript": "workspace:*"
  }
}
