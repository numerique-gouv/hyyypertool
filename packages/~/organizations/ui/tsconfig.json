{
  "compilerOptions": {
    "jsxImportSource": "hono/jsx",
    "outDir": "./node_modules/.cache/tsc"
  },
  "extends": "@~/config.typescript/api/tsconfig.json",
  "include": ["src"],
  "references": [
    {
      "path": "../../app/ui"
    },
    {
      "path": "../../app/urls"
    },
    {
      "path": "../lib"
    }
  ]
}
