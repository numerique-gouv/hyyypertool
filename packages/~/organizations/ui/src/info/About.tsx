//

import { button } from "@~/app.ui/button";
import { LocalTime } from "@~/app.ui/time/LocalTime";
import { hx_urls } from "@~/app.urls";
import type { GetFicheOrganizationByIdHandler } from "@~/organizations.lib/usecase";
import { type JSX } from "hono/jsx";
import { InactiveWarning } from "./InactiveWarning";

//

type Props = JSX.IntrinsicElements["section"] & {
  organization: Awaited<ReturnType<GetFicheOrganizationByIdHandler>>;
};

export async function About(props: Props) {
  const { organization, ...section_props } = props;
  const hx_organizations_leaders_props =
    await hx_urls.organizations.leaders.$get({
      query: { siret: organization.siret },
    });

  return (
    <section {...section_props}>
      <InactiveWarning organization={organization} />
      <ul class="list-none pl-0">
        <li>
          Creation de l'organisation :{" "}
          <b>
            <LocalTime date={organization.created_at} />
          </b>
        </li>
        <li>
          Nature juridique :{" "}
          <b>
            {organization.cached_libelle_categorie_juridique} (
            {organization.cached_categorie_juridique})
          </b>
        </li>
        <li>
          Dénomination : <b>{organization.cached_libelle}</b>
        </li>
        <li>
          Nom complet : <b>{organization.cached_nom_complet}</b>
        </li>
        <li>
          Enseigne : <b>{organization.cached_enseigne}</b>
        </li>
        <li>
          NAF/APE : <b>{organization.cached_libelle_activite_principale}</b>
        </li>
        <li>
          Tranche d'effectif :{" "}
          <b>{organization.cached_libelle_tranche_effectif}</b> (code :{" "}
          {organization.cached_tranche_effectifs}) (
          <a
            href="https://www.sirene.fr/sirene/public/variable/tefen"
            rel="noopener noreferrer"
            target="_blank"
          >
            liste code effectif INSEE
          </a>
          )
        </li>
        <li>
          État administratif : <b>{organization.cached_etat_administratif}</b> (
          <a
            href="https://www.sirene.fr/sirene/public/variable/etatAdministratifEtablissement"
            rel="noopener noreferrer"
            target="_blank"
          >
            liste état administratif INSEE
          </a>
          )
        </li>
        <li>
          Adresse : <b>{organization.cached_adresse}</b>
        </li>
        <li>
          Code officiel géographique :{" "}
          <b>{organization.cached_code_officiel_geographique}</b>
        </li>
        <li>
          Siret : <b>{organization.siret}</b> (
          <a
            href={`https://annuaire-entreprises.data.gouv.fr/entreprise/${organization.siret}`}
            rel="noopener noreferrer"
            target="_blank"
          >
            Voir la fiche annuaire entreprise de cette organisation
          </a>
          )
        </li>
      </ul>

      <div class="my-4" {...hx_organizations_leaders_props} hx-trigger="load">
        <i class="text-center">Recherche des dirigeants...</i>
      </div>

      <a
        href={`https://annuaire-entreprises.data.gouv.fr/etablissement/${organization.siret}`}
        class={button({ size: "sm", type: "tertiary" })}
        rel="noopener noreferrer"
        target="_blank"
      >
        Fiche sur l’Annuaire des Entreprises
      </a>

      <a
        href={`https://annuaire-entreprises.data.gouv.fr/dirigeants/${organization.siret.substring(0, 9)}`}
        class={button({ size: "sm", type: "tertiary" })}
        rel="noopener noreferrer"
        target="_blank"
      >
        Liste des Dirigeants - Annuaire des Entreprises
      </a>

      <details class="my-6">
        <summary>Détails de l'organisation</summary>
        <ul>
          <li>
            id : <b>{organization.id}</b>
          </li>
          <li>
            Dernière mise à jour :{" "}
            <b>
              <LocalTime date={organization.updated_at} />
            </b>
          </li>
        </ul>
      </details>
    </section>
  );
}
