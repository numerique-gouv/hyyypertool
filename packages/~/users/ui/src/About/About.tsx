//

import { z_email_domain } from "@~/app.core/schema/z_email_domain";
import { button } from "@~/app.ui/button";
import { CopyButton } from "@~/app.ui/button/components";
import { description_list } from "@~/app.ui/list";
import { urls } from "@~/app.urls";
import type { Organization } from "@~/organizations.lib/entities/Organization";
import type { GetUserInfoOutput } from "@~/users.lib/usecase/GetUserInfo";

//

type AboutProps = {
  user: GetUserInfoOutput;
  organization: Pick<Organization, "siret">;
};

//

export function About({ user, organization }: AboutProps) {
  const domain = z_email_domain.parse(user.email, { path: ["user.email"] });

  return (
    <section>
      <h3>
        <a
          class="bg-none"
          target="_blank"
          href={
            urls.users[":id"].$url({ param: { id: user.id.toString() } })
              .pathname
          }
        >
          üë®‚Äçüíª Profile
        </a>
      </h3>
      <dl class={description_list()}>
        <dt>Email </dt>
        <dd>
          {user.email}{" "}
          <CopyButton
            class="ml-2"
            text={user.email}
            variant={{ size: "sm", type: "tertiary" }}
          ></CopyButton>
        </dd>

        <dt>Domaine email </dt>
        <dd>
          {domain}{" "}
          <CopyButton
            class="ml-2"
            text={domain}
            variant={{ size: "sm", type: "tertiary" }}
          ></CopyButton>
        </dd>

        <dt>Pr√©nom </dt>
        <dd>{user.given_name}</dd>

        <dt>Nom </dt>
        <dd>{user.family_name}</dd>

        <dt>Num√©ro de t√©l√©phone </dt>
        <dd>{user.phone_number}</dd>

        <dt>Profession </dt>
        <dd>
          {user.job}{" "}
          <a
            href={`https://annuaire-entreprises.data.gouv.fr/dirigeants/${organization.siret.substring(0, 9)}`}
            class={`${button({ size: "sm", type: "tertiary" })} ml-2 mr-2 bg-white`}
            rel="noopener noreferrer"
            target="_blank"
          >
            Liste dirigeants - Annuaire entreprise API
          </a>
        </dd>
      </dl>
    </section>
  );
}
