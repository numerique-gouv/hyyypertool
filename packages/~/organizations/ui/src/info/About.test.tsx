//

import { renderHTML } from "@~/app.ui/testing";
import { expect, test } from "bun:test";
import { About } from "./About";

test("render about section", async () => {
  expect(
    await renderHTML(
      <About
        organization={{
          cached_activite_principale: "cached_activite_principale",
          cached_adresse: "cached_adresse",
          cached_categorie_juridique: "cached_categorie_juridique",
          cached_code_postal: "cached_code_postal",
          cached_est_active: true,
          cached_nom_complet: "cached_nom_complet",
          created_at: "2011-11-22 14:34:34.000Z",
          id: 42,
          siret: "siret",
          cached_code_officiel_geographique:
            "cached_code_officiel_geographique",
          updated_at: "2011-11-15T13:48:00.000Z",
          cached_libelle_activite_principale:
            "cached_libelle_activite_principale",
          cached_libelle_categorie_juridique:
            "cached_libelle_categorie_juridique",
          cached_libelle_tranche_effectif: "cached_libelle_tranche_effectif",
          cached_libelle: "cached_libelle",
          cached_enseigne: "cached_enseigne",
          cached_tranche_effectifs: "cached_tranche_effectifs",
          cached_etat_administratif: "cached_etat_administratif",
        }}
        id="about_section"
      />,
    ),
  ).toBeString();
});
