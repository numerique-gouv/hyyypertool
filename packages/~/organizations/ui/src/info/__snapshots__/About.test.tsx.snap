// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`render about section 1`] = `
"<section class="mt-6" id="about_section">
  <h3>
    <a class="bg-none" target="_blank" href="/organizations/42"
      >üèõ Organisation</a
    >
  </h3>
  <dl
    class="grid grid-cols-[222px_1fr] gap-x-3 ps-0 [&amp;_dd]:p-0 [&amp;_dd]:py-1 [&amp;_dd]:font-semibold [&amp;_dt]:border-0 [&amp;_dt]:border-r [&amp;_dt]:border-solid [&amp;_dt]:border-gray-300 [&amp;_dt]:p-0 [&amp;_dt]:py-1 [&amp;_dt]:uppercase"
  >
    <dt>D√©nomination</dt>
    <dd>
      <abbr title="cached_nom_complet">cached_libelle</abbr>
      <button
        _="
      on click
        set text to @data-text
        js(me, text)
          if (&#39;clipboard&#39; in window.navigator) {
            navigator.clipboard.writeText(text)
          }
        end
        add .copied to me
        wait 1s
        remove .copied from me"
        class="fr-btn bg-transparent text-black hover:!bg-[--background-default-grey-hover] fr-btn--sm fr-btn--tertiary ml-2"
        data-text="cached_libelle"
      >
        <span
          class="fr-icon-clipboard-line [.copied_&amp;]:hidden"
          aria-hidden="true"
        ></span
        ><span
          class="fr-icon-check-line animated bounceIn hidden [.copied_&amp;]:inline"
          aria-hidden="true"
        ></span>
      </button>
    </dd>
    <dt>Siret</dt>
    <dd>
      siret
      <a
        href="https://annuaire-entreprises.data.gouv.fr/entreprise/siret"
        class="fr-btn fr-btn--sm fr-btn--tertiary ml-2"
        rel="noopener noreferrer"
        target="_blank"
        >Fiche annuaire</a
      ><button
        _="
      on click
        set text to @data-text
        js(me, text)
          if (&#39;clipboard&#39; in window.navigator) {
            navigator.clipboard.writeText(text)
          }
        end
        add .copied to me
        wait 1s
        remove .copied from me"
        class="fr-btn bg-transparent text-black hover:!bg-[--background-default-grey-hover] fr-btn--sm fr-btn--tertiary ml-2"
        data-text="siret"
      >
        <span
          class="fr-icon-clipboard-line [.copied_&amp;]:hidden"
          aria-hidden="true"
        ></span
        ><span
          class="fr-icon-check-line animated bounceIn hidden [.copied_&amp;]:inline"
          aria-hidden="true"
        ></span>
      </button>
    </dd>
    <dt>NAF/APE</dt>
    <dd>cached_libelle_activite_principale</dd>
    <dt>Adresse</dt>
    <dd>cached_adresse</dd>
    <dt>Nature juridique</dt>
    <dd>cached_libelle_categorie_juridique (cached_categorie_juridique)</dd>
    <dt>Tranche d effectif</dt>
    <dd>
      cached_libelle_tranche_effectif (code : cached_tranche_effectifs)
      <span class="text-nowrap"
        >(<a
          href="https://www.sirene.fr/sirene/public/variable/tefen"
          rel="noopener noreferrer"
          target="_blank"
          >liste code effectif INSEE</a
        >)</span
      >
    </dd>
  </dl>
  <details class="my-6">
    <summary>D√©tails de l&#39;organisation</summary>
    <ul>
      <li>id : <b>42</b></li>
      <li>
        Cr√©ation de l&#39;organisation :
        <b
          ><time
            datetime="2011-11-22T14:34:34.000Z"
            title="Tue Nov 22 2011 15:34:34 GMT+0100 (Central European Standard Time)"
            >22/11/2011 15:34:34
          </time></b
        >
      </li>
      <li>
        Derni√®re mise √† jour :
        <b
          ><time
            datetime="2011-11-15T13:48:00.000Z"
            title="Tue Nov 15 2011 14:48:00 GMT+0100 (Central European Standard Time)"
            >15/11/2011 14:48:00
          </time></b
        >
      </li>
      <li>
        Derni√®re mise √† jour :
        <b
          ><time
            datetime="2011-11-15T13:48:00.000Z"
            title="Tue Nov 15 2011 14:48:00 GMT+0100 (Central European Standard Time)"
            >15/11/2011 14:48:00
          </time></b
        >
      </li>
    </ul>
  </details>
</section>
"
`;
